ooni/oonicli.py

```diff
@@ -315,10 +320,10 @@ def post_director_start(_):
 
             net_test_loader.annotations = global_options['annotations']
 
-            director.startNetTest(net_test_loader,
-                                    global_options['reportfile'],
-                                    collector)
-        return director.allTestsDone
+            yield director.startNetTest(net_test_loader,
+                                        global_options['reportfile'],
+                                        collector,
+                                        global_options['no-yamloo'])
```

Comment:
It's basically a way to replicate sequential code execution although it's actually asynchronous. What a yield inside of a inlineCallback does is it pauses the execution of the function until the deferred has fired, either with a callback or an errback, and resumes execution from the line below it.

The reason why the behaviour here has changed to run each measurement sequentially is that prior to this change we would schedule all the nettests at the same time.
This lead to the creation of a new report for every nettest, but since there is a limited number of slots for executing the measurements it could have happened that the created report timed out in the backend before it all being written.

By running them sequentially we ensure that we only work on one test at any given time and ensure that we keep the reports alive in the backend.